pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -141,13 +141,6 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1.ApiServerSour
 	return nil
 }
 
-func (r *Reconciler) FinalizeKind(ctx context.Context, source *v1.ApiServerSource) pkgreconciler.Event {
-	logging.FromContext(ctx).Info("Deleting source")
-	// Allow for eventtypes to be cleaned up
-	source.Status.CloudEventAttributes = []duckv1.CloudEventAttributes{}
-	return nil
-}
```

Comment:
@pierDipi I don't quite understand why we are deleting this, would you be able to explain?

Reply:
I don't get why we need this https://github.com/knative/eventing/pull/7245#discussion_r1502538737